File: frontend/src/models/Ad.ts
export type Ad = {
    id: number;
    room: {
        id: number;
        streetAddress: string;
        eirCode: string;
        roomType: string;
        ensuiteBathroom: string;
        heating: string;
        carpeted: string;
        roomImage: {
            id: number;
            name: string;
            type: string;
        }
    };
    user: {
        id: number;
        email: string;
        firstName: string;
        lastName: String;
        phoneNo: string;

    };
    rent: number;
    billsIncluded: string;
    parking: string;
    ownerOccupied: string;
    petAllowed: string;
    washingMachine: string;
    dryer: string;
    dishWasher: string;
    createdAt: string;
    roomImageId: number;
   
}

File: backend/src/main/java/com/caiotayota/findmyroom/entities/Ad.java
package com.caiotayota.findmyroom.entities;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.Date;

@Entity
@Table(name = "ads")
@Getter
@Setter
@NoArgsConstructor
public class Ad {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ad_id")
    private Long id;

    @OneToOne
    @JoinColumn(name = "room_id")
    private Room room;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    private BigDecimal rent;
    private boolean billsIncluded;
    private boolean ownerOccupied;
    private boolean parking;
    private boolean petAllowed;
    private boolean washingMachine;
    private boolean dryer;
    private boolean dishWasher;

    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;

    @Temporal(TemporalType.TIMESTAMP)
    @UpdateTimestamp
    private Date updatedAt;

}
